src/lib/authserver/dir/ds389.rb

```diff
@@ -37,29 +37,42 @@ def self.get_instance_names
   end
 
   # gen_setup_ini generates INI file content with parameters for setting up directory server.
-  def self.gen_setup_ini(fqdn, instance_name, suffix, dm_dn, dm_pass)
-    return "[General]
-FullMachineName=#{fqdn}
-SuiteSpotUserID=dirsrv
-SuiteSpotGroup=dirsrv
+  def self.gen_setup_ini(fqdn, instance_name, suffix, dm_pass)
+    return "# Generated by yast-auth-server
+[general]
+config_version = 2
+full_machine_name = #{fqdn}
+# This may be need to be tweaked, it could break setups ...
+# strict_host_checking = true/false
 
 [slapd]
-ServerPort=389
-ServerIdentifier=#{instance_name}
-Suffix=#{suffix}
-RootDN=#{dm_dn}
-RootDNPwd=#{dm_pass}
-AddSampleEntries=No
+root_password = #{dm_pass}
+instance_name = #{instance_name}
+
+[backend-userroot]
+sample_entries = yes
+suffix = #{suffix}
 "
   end
 
   # exec_setup runs setup-ds.pl using input parameters file content.
-  # The output of setup script is written into file /root/yast2-auth-server-dir-setup.log
+  # The output of setup script is written into file .y2log or /var/log/YaST/y2log
   # Returns true only if setup was successful.
   def self.exec_setup(content)
+    append_to_log('Beginning YAST auth server installation ...')
+
     open(DS_SETUP_INI_PATH, 'w') {|fh| fh.puts(content)}
-    stdin, stdouterr, result = Open3.popen2e('/usr/sbin/setup-ds.pl', '--debug', '--silent', '-f', DS_SETUP_INI_PATH)
-    append_to_log(stdouterr.readlines.join('\n'))
+    # dry run first to see if it breaks ...
+    stdin, stdouterr, result = Open3.popen2e('/usr/sbin/dscreate', '-v', 'from-file', '-n', DS_SETUP_INI_PATH)
+    stdouterr.readlines.map { |l| append_to_log(l) }
+
+    if result.value.exitstatus != 0
```

Comment:
above your append to log do only debug logging. So in case of failure. I really suggest here to log to info or warn when customers report a bug, it is useful.